pkg/pillar/worker/worker.go

```diff
@@ -17,15 +18,20 @@ import (
 type Worker struct {
 	// Private
 	requestChan  chan<- Work
-	resultChan   <-chan privateResult
+	resultChan   <-chan Processor
 	requestCount uint // Number of work items submitted
 	resultCount  uint // Number of work results processed
+	workMap      map[string]bool
+	resultMap    map[string]WorkResult
+	handlers     map[string]Handler
 }
 
 // Work is one work item
 // Key is used to match with the WorkResult
 // The description is specific for the particular worker function
+// and is passed as is to the worker
 type Work struct {
+	Kind        string
```

Comment:
Is Kind required? A comment would be helpful, plus enforcement if Kind can not be empty.

Reply:
Sure thing. Adding.